File: hreso/src/main/cpp/transform/imageoptionparams.cpp
//
// Created by cangwang on 2022/10/14.
//

#include "imageoptionparams.h"


void ImageOptionParams::setOptions(string options, jobject op) {
    if (options.empty()) {
        HLOGE("option is null");
        return;
    }
    this->options = options;
    char g_log_info[200];

    JSON_Value *root_value = json_parse_string(options.c_str());
    // 获取根类型 object/array
    JSON_Value_Type jsonValueType = json_value_get_type(root_value);
    if (jsonValueType == JSONObject) {
        JSON_Object *jsonObject = json_value_get_object(root_value);
        //解析descript
        type = (int)json_object_get_number(jsonObject, "type");
        name = json_object_get_string(jsonObject, "name");
        filterType = json_object_get_string(jsonObject, "filterType");
        address = json_object_get_string(jsonObject, "address");
        saveAddress = json_object_get_string(jsonObject, "saveAddress");
        scaleType = (int)json_object_get_number(jsonObject, "type");
        scaleRatio = (float)json_object_get_number(jsonObject, "scaleRatio");
        scaleWidth = (int)json_object_get_number(jsonObject, "scaleWidth");
        scaleHeight = (int)json_object_get_number(jsonObject, "scaleHeight");
        front = json_object_get_boolean(jsonObject, "front");
        async = json_object_get_boolean(jsonObject, "async");
    }
    this->op = op;
}

File: hreso/src/main/java/com/cangwang/hreso/bean/ImageOptionParams.kt
package com.cangwang.hreso.bean

import org.json.JSONObject

/**
 * image setting option
 */
class ImageOptionParams {
    var name = ""
    var address = ""
    var saveAddress = ""
    var filterType = "fsr"  //滤镜名称
    var type = 1  // 1是图片 2是视频
    var scaleType = 1  //1 使用scaleRatio, 2使用固定宽高
    var scaleRatio = 1f //缩放倍数
    var scaleWidth = 0f  //缩放后的宽
    var scaleHeight = 0f  //缩放后的高
    var rotation = 0   //图片角度
    var front = false //是否插入队头
    var async = false //是否同步转换

    fun toJson(): String {
        val json = JSONObject()
        json.put("name", name)
        json.put("address", address)
        json.put("saveAddress", saveAddress)
        json.put("type", type)
        json.put("filterType", filterType)
        json.put("scaleType", scaleType)
        json.put("scaleRatio", scaleRatio)
        json.put("scaleWidth", scaleWidth)
        json.put("scaleHeight", scaleHeight)
        json.put("rotation", rotation)
        json.put("front", front)
        json.put("async", async)
        return json.toString()
    }

    fun parse(option: String) {
        val json = JSONObject(option)
        name = json.optString("name")
        address = json.optString("address")
        saveAddress = json.optString("saveAddress")
        filterType = json.optString("filterType")
        type = json.optInt("type")
        scaleType = json.optInt("scaleType")
        scaleRatio = json.optDouble("scaleRatio").toFloat()
        scaleWidth = json.optDouble("scaleRatio").toFloat()
        scaleHeight = json.optDouble("scaleHeight").toFloat()
        front = json.optBoolean("front")
        async = json.optBoolean("async")
    }

    override fun toString(): String {
        return "OptionParams(name='$name', address='$address', saveAddress='$saveAddress', " +
                "filterType='$filterType', type=$type, scaleType=$scaleType, " +
                "scaleRatio=$scaleRatio, scaleWidth=$scaleWidth, scaleHeight=$scaleHeight, " +
                "rotation = $rotation, front=$front, async=$async)"
    }
}